### PR TITLE
Basic error handling for yaml.

### PR DESCRIPTION
**Pull request by robertwb**
This is actuated by a error_handling clause.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by robertwb** [comment]
R: @bzablocki

**Comment by robertwb** [comment]
Run Python PreCommit

**Comment by robertwb** [comment]
retest this please

**Review by bzablocki — COMMENTED** [review]
I left a few comments, thanks!

File: sdks/python/apache_beam/transforms/core.py [hunk 1]
```diff
@@ -2182,6 +2185,9 @@ def expand(self, pcoll):
         *self._args,
         **self._kwargs).with_outputs(
             self._dead_letter_tag, main=self._main_tag, allow_unknown_tags=True)
+    #TODO(BEAM-18957): Fix when type inference supports tagged outputs.
+    result[self._main_tag].element_type = self._original_infer_element_type(
+        pcoll.element_type)
```
[the remark this answers is not in the document]
Comment (robertwb) [reply sdks/python/apache_beam/transforms/core.py:2190]: Yes, we could do that. `infer_output_type` is overridden by WindowInto, but an implementation is easily enough added to the implementing DoFn itself.

File: sdks/python/apache_beam/transforms/core.py [hunk 2]
```diff
@@ -2244,6 +2250,82 @@ def process(self, *args, **kwargs):
                   traceback.format_exception(*sys.exc_info()))))
 
 
+# Idea adapted from Asguard.
```
[the remark this answers is not in the document]
Comment (robertwb) [reply sdks/python/apache_beam/transforms/core.py:2253]: Yes. Fixed (added reference).

File: sdks/python/apache_beam/transforms/core.py [hunk 3]
```diff
@@ -2244,6 +2250,82 @@ def process(self, *args, **kwargs):
...
+    return _PValueWithErrors(
+        result[self.main_output_tag()],
+        self._exception_handling_args,
+        self._upstream_errors + (result[self.error_output_tag()], ))
+
+  def accumulated_errors(self):
+    if len(self._upstream_errors) == 1:
+      return self._upstream_errors[0]
+    else:
+      return self._upstream_errors | Flatten()
```
[the remark this answers is not in the document]
Comment (robertwb) [reply sdks/python/apache_beam/transforms/core.py:2291]: Maybe, but that might have issues with streaming pipeline upgrades. Keeping as is for now as not all runners handle one-element flattens optimally.

**Review by bzablocki — APPROVED** [review]
LGTM, thanks!

**Comment by robertwb** [comment]
apache_beam.io.gcp.bigquery_test.PipelineBasedStreamingInsertTest.test_batch_size_with_auto_sharding_0 looks unrelated.

**Comment by robertwb** [comment]
Run Python PreCommit

**Pull request merged**